src/Core/AbstractApi.php

```diff
@@ -97,7 +97,7 @@ public function __construct($configuration = [], ?CredentialProvider $credential
      */
     final protected function getResponse(string $method, $body, $headers = [], ?string $endpoint = null): ResponseInterface
     {
-        $body = $this->stringifyBody($body);
+        $body = $this->stringifyBody($body); // can't we keep the streaming upload behavior?
```

Comment:
No.. and it is super weird. We need the body for signing the request. This is how the official SDK is doing to...